core/src/main/scala/kafka/log/Log.scala

```diff
@@ -244,7 +244,8 @@ class Log(@volatile private var _dir: File,
           val producerIdExpirationCheckIntervalMs: Int,
           val topicPartition: TopicPartition,
           val producerStateManager: ProducerStateManager,
-          logDirFailureChannel: LogDirFailureChannel) extends Logging with KafkaMetricsGroup {
+          logDirFailureChannel: LogDirFailureChannel,
+          val hadCleanShutdown: Boolean = true) extends Logging with KafkaMetricsGroup {
```

Comment:
Could we add the new param to the javadoc above?

Reply:
Done